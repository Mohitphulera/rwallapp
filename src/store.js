import { createStore } from 'redux'

const initialState = {
    user: null
}

const SET_USER = 'SET_USER'

const reducer = (state = initialState, action) => {
    switch(action.type) {
        case SET_USER: {
            console.log(action.payload ? 'logging in' : 'logging out')
            return {
                user: action.payload
            }
        }
        default: {
            return state
        }
    }
}


export function setUser(user = null) {
    return {
        type: SET_USER,
        payload: user
    }
}

export const store = createStore(reducer)