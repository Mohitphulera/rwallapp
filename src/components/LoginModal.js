import React, { Component } from 'react'
import { connect } from 'react-redux'
import { request, setToken } from '../services/request'
import { setUser } from '../store'
import "./style.css"
import loginImg from "../../src/logo.png";


const initialState = {
    username: '',
    password: '',
    loading: false,
    error: false
}

class LoginModal extends Component {
    constructor(props) {
        super(props)
        this.state = initialState
        this.timeout = null
    }

    componentWillUnmount() {
        if (this.timeout) {
            clearTimeout(this.timeout)
        }
    }

    render() {
        const { loading, error } = this.state
        const { user } = this.props
        let modalCard = null
        if (!user) {
            modalCard = (
                <div className="modal-card">
                    <header className="modal-card-head">
                        <p className="modal-card-title">Login</p>
                        <button className="delete" onClick={() => this.close()} aria-label="close" />
                    </header>
                    <section className="modal-card-body">
                        <form>
                        <img  className="logo" src={loginImg} alt="Tsl" />
                            <div className="field">
                                <label className="label">Username</label>
                                <div className="control has-icons-left has-icons-right">
                                    <input
                                        value={this.state.username}
                                        onChange={({ target: { value } }) => this.setState({ username: value })}
                                        onKeyPress={e => (e.key === 'Enter' ? this.submit() : null)}
                                        name="username"
                                        className="input"
                                        type="text"
                                        required
                                    />
                                    <span className="icon is-small is-left">
                                        <i className="fa fa-user" />
                                    </span>
                                </div>
                            </div>
                            <div className="field">
                                <label className="label">Password</label>
                                <div className="control has-icons-left has-icons-right">
                                    <input
                                        value={this.state.password}
                                        onChange={({ target: { value } }) => this.setState({ password: value })}
                                        onKeyPress={e => (e.key === 'Enter' ? this.submit() : null)}
                                        name="password"
                                        className="input"
                                        type="password"
                                        required
                                    />
                                    <span className="icon is-small is-left">
                                        <i className="fa fa-lock" />
                                    </span>
                                </div>
                            </div>
                        </form>
                    </section>
                    <footer className="modal-card-foot">
                        <button onClick={() => this.submit()} className="button is-link">
                            <span>Submit</span>
                            {loading ? (
                                <span className="icon">
                                    <i className="fa fa-circle-o-notch fa-spin" />
                                </span>
                            ) : null}
                        </button>
                        <button className="button" onClick={() => this.close()}>
                            Cancel
                        </button>
                        {error ? <p className="help is-danger">Invalid login information.</p> : null}
                    </footer>
                </div>
            )
        } else {
            modalCard = (
                <div className="modal-card">
                    <header className="modal-card-head">
                        <p className="modal-card-title">Welcome back!</p>
                        <button className="delete" onClick={() => this.close()} aria-label="close" />
                    </header>
                    <section className="modal-card-body">
                        <div className="content">Glad to see you again, {this.state.username}!</div>
                    </section>
                    <footer className="modal-card-foot" />
                </div>
            )
        }

        return (
            <div className="modal is-active">
                <div className="modal-background" onClick={() => this.close()} />
                {modalCard}
            </div>
        )
    }

    submit() {
        this.setState({ loading: true })
        let user = {
            username: this.state.username,
            password: this.state.password
        }
        request
            .post('auth/', user)
            .then(data => {
                if (data && data.token) {
                    setToken(data.token)
                    this.props.login({
                        ...user,
                        token: data.token
                    })
                    // Reset component state
                    this.setState({
                        error: false,
                        loading: false
                    })
                    // Auto close
                    this.timeout = setTimeout(() => {
                        this.props.close()
                    }, 3000)
                }
            })
            .catch(() => {
                this.setState({
                    loading: false,
                    error: true
                })
            })
    }

    close() {
        if (!this.state.loading) {
            this.props.close()
        }
    }
}

const mapStateToProps = state => ({ user: state.user })
const mapDispatchToProps = dispatch => ({ login: user => dispatch(setUser(user)) })
export default connect(
    mapStateToProps,
    mapDispatchToProps
)(LoginModal)
