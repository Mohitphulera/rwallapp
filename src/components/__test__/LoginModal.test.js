import React from 'react'
import ReactDom from 'react-dom'
import LoginModal from './../LoginModal'
import {isTSAnyKeyword} from  '@babel/types'
import { Provider } from 'react-redux'
import { store } from './../../store'
import renderer from "react-test-renderer"

it("renders without crashing",()=> {
    const div = document.createElement("div");
    ReactDom.render( <Provider store={store}>
        <LoginModal />
      </Provider>,div);

})

it("matches snapshot & renders correctly",() => {
    const tree=renderer.create(<Provider store={store}>
        <LoginModal />
      </Provider>).toJSON();
    expect(tree).toMatchSnapshot();
})