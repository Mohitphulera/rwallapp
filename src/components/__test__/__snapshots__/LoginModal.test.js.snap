// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot & renders correctly 1`] = `
<div
  className="modal is-active"
>
  <div
    className="modal-background"
    onClick={[Function]}
  />
  <div
    className="modal-card"
  >
    <header
      className="modal-card-head"
    >
      <p
        className="modal-card-title"
      >
        Login
      </p>
      <button
        aria-label="close"
        className="delete"
        onClick={[Function]}
      />
    </header>
    <section
      className="modal-card-body"
    >
      <form>
        <img
          alt="Tsl"
          className="logo"
          src="logo.png"
        />
        <div
          className="field"
        >
          <label
            className="label"
          >
            Username
          </label>
          <div
            className="control has-icons-left has-icons-right"
          >
            <input
              className="input"
              name="username"
              onChange={[Function]}
              onKeyPress={[Function]}
              required={true}
              type="text"
              value=""
            />
            <span
              className="icon is-small is-left"
            >
              <i
                className="fa fa-user"
              />
            </span>
          </div>
        </div>
        <div
          className="field"
        >
          <label
            className="label"
          >
            Password
          </label>
          <div
            className="control has-icons-left has-icons-right"
          >
            <input
              className="input"
              name="password"
              onChange={[Function]}
              onKeyPress={[Function]}
              required={true}
              type="password"
              value=""
            />
            <span
              className="icon is-small is-left"
            >
              <i
                className="fa fa-lock"
              />
            </span>
          </div>
        </div>
      </form>
    </section>
    <footer
      className="modal-card-foot"
    >
      <button
        className="button is-link"
        onClick={[Function]}
      >
        <span>
          Submit
        </span>
      </button>
      <button
        className="button"
        onClick={[Function]}
      >
        Cancel
      </button>
    </footer>
  </div>
</div>
`;
